tests/integration/ceph_mgr_test.go

```diff
@@ -0,0 +1,216 @@
+)
+
+// **************************************************
+// *** Mgr operations covered by TestMgrSmokeSuite ***
+//
+// Ceph orchestrator device ls
+// Ceph orchestrator status
+// Ceph orchestrator host ls
+// Ceph orchestrator create OSD
+// Ceph orchestrator ls
+// **************************************************
+func TestMgrSmokeSuite(t *testing.T) {
+	if installer.SkipTestSuite(installer.CephTestSuite) {
+		t.Skip()
+	}
+
+	s := new(MgrSmokeSuite)
+	defer func(s *MgrSmokeSuite) {
+		HandlePanics(recover(), s.op, s.T)
+	}(s)
+	suite.Run(t, s)
+}
+
+type MgrSmokeSuite struct {
+	suite.Suite
+	op        *TestCluster
+	k8sh      *utils.K8sHelper
+	namespace string
+	CLI       MgrCLI
+}
+
+type host struct {
+	Addr     string
+	Hostname string
+	Labels   []string
+	Status   string
+}
+
+type service struct {
+	Container_image_name string
+	Service_name         string
+	Size                 int
+	Running              int
+	Spec                 interface{}
+	Last_refresh         string
+}
+
+func (suite *MgrSmokeSuite) SetupSuite() {
+	suite.namespace = "mgr-ns"
+
+	mgrTestCluster := TestCluster{
+		namespace:        suite.namespace,
+		storeType:        "bluestore",
+		storageClassName: "",
+		useHelm:          false,
+		usePVC:           false,
+		mons:             3,
+		rbdMirrorWorkers: 1,
+		rookCephCleanup:  true,
+		withoutOSDs:      true,
+	}
+
+	suite.op, suite.k8sh = StartTestCluster(suite.T, smokeSuiteMinimalTestVersion, &mgrTestCluster, installer.VersionMaster, installer.MasterVersion)
+
+	suite.CLI = MgrCLI{k8sh: suite.k8sh, namespace: suite.namespace}
+
+}
+
+func (suite *MgrSmokeSuite) AfterTest(suiteName, testName string) {
+	suite.op.installer.CollectOperatorLog(suiteName, testName, installer.SystemNamespace(suite.namespace))
+}
+
+func (suite *MgrSmokeSuite) TearDownSuite() {
+	suite.op.Teardown()
+}
+
+func (suite *MgrSmokeSuite) TestDeviceLs() {
+	logger.Infof("Testing .... <orch device ls>")
+	err, device_list := suite.CLI.Execute([]string{"device", "ls"})
+	assert.Nil(suite.T(), err)
+
+	logger.Infof("device ls RESULT =%v", device_list)
+}
+
+func (suite *MgrSmokeSuite) TestStatus() {
+	logger.Infof("Testing .... <orch status>")
+	err, status := suite.CLI.Execute([]string{"status"})
+	assert.Nil(suite.T(), err)
+	assert.Equal(suite.T(), status, "Backend: rook\nAvailable: True")
+}
+
+func (suite *MgrSmokeSuite) TestHostLs() {
+	logger.Infof("Testing .... <orch host ls>")
+	// Get the orchestrator hosts
+	err, output := suite.CLI.Execute([]string{"host", "ls", "json"})
+	assert.Nil(suite.T(), err)
+
+	hosts := []byte(output)
+	var hostsList []host
+
+	err = json.Unmarshal(hosts, &hostsList)
+	if err != nil {
+		assert.Nil(suite.T(), err)
+	}
+
+	var hostOutput []string
+	for _, hostItem := range hostsList {
+		hostOutput = append(hostOutput, hostItem.Addr)
+	}
+	sort.Strings(hostOutput)
+
+	// get the k8s nodes
+	nodes, err := suite.k8sh.Kubectl("get", "nodes", "-o=custom-columns=NAME:.metadata.name", "--no-headers")
```

Comment:
Where possible it's preferred to call the k8s API instead of the `kubectl` commands. The test code is more difficult to maintain that way.  In this case you could probably call rook's helper `k8sutil.GetNodeHostNames()`

Reply:
Ok. Removed all the kubectl calls